.github/actions/mirror-branches/action.yml

```diff
@@ -0,0 +1,182 @@
+      Required PAT to fetch the destination repository's branches and to push to it.
+    required: true
+  branch-names:
+    description: >
+      A comma-separated list of branch names to push to the destination repository. Use this parameter when the names of
+      the branches that are outdated on the destination are known, e.g. 'github.ref_name' when the workflow is triggered
+      by the push event in the source repository. This parameter takes precedence over 'branch-regex' when not empty.
+      Example: 'dev, feature/test, issue/TEST-1234'.
+  branch-regex:
+    description: >
+      A regex to match the branches of the source repository. Matching branches' head commit hashes will be compared
+      between the source and destination repository using the GitHub API to prevent unnecessary push operations. Default
+      value matches any branch. Example: '^dev$|^issue\/.+$' mirrors the 'dev' branch and the ones starting with
+      'issue/'.
+    default: .+
+
+runs:
+  using: composite
+  steps:
+    - name: Parameter validation
+      shell: pwsh
+      run: |
+        $branchNames = '${{ inputs.branch-names }}'.Trim()
+        $branchRegex = '${{ inputs.branch-regex }}'.Trim()
+        $sourceRepository = '${{ inputs.source-repository }}'
+        $destinationRepository = '${{ inputs.destination-repository }}'
+
+        if (-not $branchNames -and -not $branchRegex)
+        {
+          throw 'At least one of the ''branch-names'' and ''branch-regex'' parameters must be non-empty.'
+        }
+
+        if (-not $sourceRepository -match '^[^/]+/[^/]+$')
+        {
+          throw 'Please provide a valid source repository path.'
+        }
+
+        if (-not $destinationRepository -match '^[^/]+/[^/]+$')
+        {
+          throw 'Please provide a valid destination repository path.'
+        }
+
+        if ($sourceRepository.ToLower() -eq $destinationRepository.ToLower())
+        {
+          throw 'The source and the destination repositories cannot be the same.'
+        }
+
+    - name: Setup Scripts
+      shell: pwsh
+      run: |
+        '${{ github.action_path }}' >> $Env:GITHUB_PATH
+        (Resolve-Path '${{ github.action_path }}/../../../Scripts').Path >> $Env:GITHUB_PATH
+
+    - name: Get source branches
+      id: get-source-branches
+      uses: Lombiq/GitHub-Actions/.github/actions/get-branches@issue/OSOE-737
+      with:
+        repository: ${{ inputs.source-repository }}
+        # When empty, source token falls back to destination token, which still works for a public source repository.
+        api-token: ${{ inputs.source-token == '' && inputs.destination-token || inputs.source-token }}
+
+    - name: Get destination branches
+      id: get-destination-branches
+      uses: Lombiq/GitHub-Actions/.github/actions/get-branches@issue/OSOE-737
+      with:
+        repository: ${{ inputs.destination-repository }}
+        api-token: ${{ inputs.destination-token }}
+
+    - name: Find branches to mirror
```

Comment:
Do I understand correctly that to prevent empty pushes, the logic here is to poll the branches of the source repo and compare if their head commits equal to the head commits of the corresponding destination repo branches?

Reply:
Great, thanks!